File: 150. Evaluate Reverse Polish Notation/main.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in range(len(tokens)):
            if tokens[i] not in ('+', '-', '*', '/'):
                stack.append(int(tokens[i]))
            else:
                if tokens[i] == '+':
                    stack.append(int(stack.pop()) + int(stack.pop()))
                elif tokens[i] == '-':
                    pas = int(stack.pop())
                    stack.append(int(stack.pop()) - pas)
                elif tokens[i] == '*':
                    stack.append(int(stack.pop()) * int(stack.pop()))
                else:
                    pas = int(stack.pop())
                    stack.append(int(int(stack.pop()) / pas))
        return stack.pop()

File: 150. Evaluate Reverse Polish Notation/main.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
  let stack = [];

  for (let i = 0; i < tokens.length; i++) {
    if (tokens[i] == "+") {
      num1 = stack.pop();
      num2 = stack.pop();
      stack.push(num2 + num1);
    } else if (tokens[i] == "-") {
      num1 = stack.pop();
      num2 = stack.pop();
      stack.push(num2 - num1);
    } else if (tokens[i] == "*") {
      num1 = stack.pop();
      num2 = stack.pop();
      stack.push(num2 * num1);
    } else if (tokens[i] == "/") {
      num1 = stack.pop();
      num2 = stack.pop();
      stack.push(parseInt(num2 / num1));
    } else {
      stack.push(parseInt(tokens[i]));
    }
  }

  return stack.pop();
};
